src/Microsoft.AspNet.OData.Shared/EnableQueryAttribute.cs

```diff
@@ -509,7 +514,20 @@ public virtual object ApplyQuery(object entity, ODataQueryOptions queryOptions)
                 throw Error.InvalidOperation(SRResources.QueryGetModelMustNotReturnNull);
             }
 
-            return new ODataQueryContext(model, elementClrType, path);
+            ODataQueryContext queryContext = new ODataQueryContext(model, elementClrType, path);
```

Comment:
In Core version,  to use "ODataFeature" to add the Per-request information.
In Classic version, to use "ODataProperties" to add the Per-request information.